pkg/relayer/session/session_test.go

```diff
@@ -73,9 +73,10 @@ func requireProofCountEqualsExpectedValueFromProofParams(t *testing.T, proofPara
 	}
 	supplierOperatorAddress := sample.AccAddress()
 	// Set the supplier operator balance to be able to submit the expected number of proofs.
-	claimAndFeeGasCost := session.ClamAndProofGasCost.Amount.Int64()
-	numExpectedProofs := int64(2)
-	supplierOperatorBalance := claimAndFeeGasCost * numExpectedProofs
+	feePerProof := prooftypes.DefaultParams().ProofSubmissionFee.Amount.Int64()
+	gasCost := session.ClamAndProofGasCost.Amount.Int64()
+	proofCost := feePerProof + gasCost
+	supplierOperatorBalance := proofCost
 	supplierClientMap := testsupplier.NewClaimProofSupplierClientMap(ctx, t, supplierOperatorAddress, proofCount)
 	blockPublishCh, minedRelaysPublishCh := setupDependencies(t, ctx, supplierClientMap, emptyBlockHash, proofParams, supplierOperatorBalance)
```

Comment:
Can you do a comparison of the performance before and update adding a cache? Similar to what you did with the SMT.

E.g. I don't know how to (easily) to a load test on DevNet so anything like:
- Makefile
- E2E test
- Docs / instructions 

Would be really nice!

Reply:
```
// TODO_TECHDEBT: Consider a flag to change client queriers caching behavior.
// This would allow to easily switch between caching and non-caching queriers
// for benchmarking purposes.
```